go/common/host/host.go

```diff
@@ -46,3 +50,12 @@ type P2P interface {
 	UpdatePeerList([]string)
 	BroadcastTx(tx common.EncryptedTx) error
 }
+
+// BlockProvider interface allows host to monitor and await L1 blocks for feeding to enclave
+type BlockProvider interface {
+	StartStreamingFromHeight(height *big.Int) (<-chan *types.Block, error)
```

Comment:
is this supposed to be a re-connecting stream?
If not, who's responsibility is it to handle reconnections